### PR TITLE
Resolve #198, #200, #202

### LINKED ISSUE #198: Transfer eth to escrow address for ledger ERC20 tx

### LINKED ISSUE #200: Re-select wallet cause app to crash

### PR DESCRIPTION
**Pull request by InVincible2016**
- Resolve #198: send some eth as tx fee for future tx before sending ERC20 tokens

- Resolve #200 re-select wallet cause app to crash

### CODE REVIEW

File: src/actions/transferActions.js [hunk 1]
```diff
@@ -202,8 +202,32 @@ async function _submitTx (dispatch, txRequest, accountInfo) {
...
-      const amountInWei = _web3.utils.toWei(transferAmount.toString(), 'ether')
-      const signedTransactionObject = await ledgerNanoS.signSendTrasaction(0, getCrypto('dai').address, ERC20_ABI, 'transfer', escrow.address, amountInWei)
+      // send out tx fee for next tx
+      const signedTxFee = await ledgerNanoS.signSendEther(
+        0,
+        escrow.address,
+        txCost.costByType.ethTransfer,
+        {
+          gasLimit: parseInt(txCost.costByType.txCostEth.gas),
+          gasPrice: parseInt(txCost.costByType.txCostEth.price)
```
Comment (timothywangdev) [inline src/actions/transferActions.js]: potential overflow, consider using string

File: src/actions/transferActions.js [hunk 2]
```diff
@@ -202,8 +202,32 @@ async function _submitTx (dispatch, txRequest, accountInfo) {
...
+        })
+      txRequest.sendTxFeeTxHash = await new Promise((resolve, reject) => {
```
Comment (timothywangdev) [inline src/actions/transferActions.js]: consider using https://github.com/ventureum/Chainsfer/blob/fcac2c23181638a3d2b133491f182fc642f26bdd/src/actions/transferActions.js#L24

File: src/actions/transferActions.js [hunk 3]
```diff
@@ -202,8 +202,32 @@ async function _submitTx (dispatch, txRequest, accountInfo) {
...
+      const amountInBasicUnit = _web3.utils.toWei(transferAmount.toString(), 'ether')
+      const signedTransactionObject = await ledgerNanoS.signSendTrasaction(
+        0,
+        getCrypto('dai').address, ERC20_ABI,
+        'transfer',
+        escrow.address,
+        amountInBasicUnit,
+        {
+          gasLimit: parseInt(txCost.costByType.txCostERC20.gas),
+          gasPrice: parseInt(txCost.costByType.txCostERC20.price)
```
Comment (timothywangdev) [inline src/actions/transferActions.js]: potential overflow

File: src/actions/transferActions.js [hunk 4]
```diff
@@ -202,8 +202,32 @@ async function _submitTx (dispatch, txRequest, accountInfo) {
...
+        })
       _web3.eth.sendSignedTransaction(signedTransactionObject.rawTransaction)
```
Comment (timothywangdev) [inline src/actions/transferActions.js]: consider using https://github.com/ventureum/Chainsfer/blob/fcac2c23181638a3d2b133491f182fc642f26bdd/src/actions/transferActions.js#L24

File: src/ledgerSigner/index.js [hunk 5]
```diff
@@ -134,7 +132,7 @@ class LedgerNanoS {
     const web3 = this.getWeb3()
     const ethLedger = await this.getEtherLedger()
     const address = await this.getEthAddress(accountIndex)
-    const txCount = await web3.eth.getTransactionCount(address)
+    const txCount = await web3.eth.getTransactionCount(address, 'pending')
```
Comment (timothywangdev) [inline src/ledgerSigner/index.js]: what's pending doing here?

File: src/ledgerSigner/index.js [hunk 6]
```diff
@@ -222,28 +219,29 @@ class LedgerNanoS {
...
+    if (gasNeeded > gasLimit) {
       console.error('Insufficient gas set for transaction.')
     } else if (gasLimit === undefined) {
       gasLimit = gasNeeded
     }
 
     let rawTx = {
       from: address,
-      nonce: txCount,
+      nonce: txCount + 1,
```
Comment (timothywangdev) [inline src/ledgerSigner/index.js]: inconsistent with line #49

File: src/containers/WalletSelectionContainer.jsx [hunk 7]
```diff
@@ -26,7 +26,7 @@ class WalletSelectionContainer extends Component {
 
   componentDidUpdate (prevProps) {
     const { wallet, cryptoSelection, syncLedgerAccountInfo, actionsPending, walletSelection } = this.props
-    if (wallet.connected && !wallet.crypto[cryptoSelection] && !actionsPending.syncAccountInfo && walletSelection === 'ledger') {
+    if (wallet && wallet.connected && !wallet.crypto[cryptoSelection] && !actionsPending.syncAccountInfo && walletSelection === 'ledger') {
```
Comment (timothywangdev) [inline src/containers/WalletSelectionContainer.jsx]: line too long

File: src/actions/transferActions.js (hunk 2 above)
Comment (InVincible2016) [reply src/actions/transferActions.js]: it uses difference web3 functions.

**Pull request merged**